src/twisted/scripts/trial.py

```diff
@@ -36,6 +36,39 @@
 }
 
 
+def _autoJobs() -> int:
+    """
+    Heuristically guess the number of job workers to run.
+
+    When ``os.process_cpu_count()`` is available (Python 3.13+),
+    return the number of logical CPUs usable by the current
+    process. This respects the ``PYTHON_CPU_COUNT`` environment
+    variable and/or ``python -X cpu_count`` flag.
+
+    Otherwise, if ``os.sched_getaffinity()`` is available (on some
+    Unixes) this returns the number of CPUs this process is
+    restricted to, under the assumption that this affinity will
+    be inherited.
+
+    Otherwise, consult ``os.cpu_count()`` to get the number of
+    logical CPUs.
+
+    Failing all else, return 1.
+
+    @returns: A strictly positive integer.
+    """
+    number: Optional[int]
+    if getattr(os, "process_cpu_count", None) is not None:
+        number = os.process_cpu_count()  # type: ignore[attr-defined]
+    elif getattr(os, "sched_getaffinity", None) is not None:
```

Comment:
Can [sched_getaffinity](https://docs.python.org/3/library/os.html#os.sched_getaffinity) return None ?

Reply:
Not as documented, though I haven't audited the implementation. There is test coverage of it returning an empty set (though that is probably impossible?)